nir/src/main/scala/scala/scalanative/nir/serialization/BinaryDeserializer.scala

```diff
@@ -234,6 +234,7 @@ final class BinaryDeserializer(buffer: ByteBuffer) {
     case T.VoidType        => Type.Void
     case T.VarargType      => Type.Vararg
     case T.PtrType         => Type.Ptr
+    case T.SizeType        => sizeType
```

Comment:
I was considering that, but would that cause issues with link-time evaluation of math expressions where `Type.Size` would not be available for calculations?

Reply:
Type.Size is special because it's effectively a type alias that we need to resolve at link time (rather than compile-time like in scalac). There is no machinery to dealias those kind of things atm, but I don't think putting platform-dependant logic into the deserializer is a good idea. Currently it's just 1-to-1 mapping without any logic whatsoever, it should rather stay this way.

Reply:
Actually if we had a way to dealias at link-time we wouldn't need separate artifacts for 32-bit and 64-bit platforms. We could have one NIR version of everything that's automagically become 32-bit or 64-bit based on the requested target architecture.

Reply:
But I think it's better left for next PR(s), it's likely not trivial and needs some design discussion as well.

Reply:
I ended up being able to eliminate `Type.Size` altogether, actually, by figuring out what the `Word` type is in the compiler plugin. Especially since native libraries sometimes use types that differ across 32/64bit systems, I think link-time switching won't work since it wouldn't be possible to express these differing APIs.